ipapython/ipautil.py

```diff
@@ -1037,6 +1037,53 @@ def host_port_open(host, port, socket_type=socket.SOCK_STREAM,
     return port_open
 
 
+def host_port_free(host, port, socket_type=socket.SOCK_STREAM,
```

Comment:
Why are you introducing a new function. We already have a ``host_port_open`` helper That looks very similar to your code.

Reply:
Please see my comments here: https://pagure.io/freeipa/issue/7415

You cannot negate the result of host_port_open() to determine if a port is free

Reply:
I left a comment on the ticket. We need a different approach to check if a port is bindable. Even the new check doesn't work properly.